### ISSUE TITLE
Databaser.py: I have a problem

### ISSUE THREAD

**Issue by germant13** [DittmarLab/HGTector#12]
hi!!,
I met a problem when creating a batadase, it show the error like that:

germanmtraglia@ubuntu:/media/germanmtraglia/6E5CA6935CA6561D/seq/new_seq_robert/HGTector-master/scripts$ python databaser.py 

Databaser: generate taxonomically balanced, non-redundant proteome database.
Author: Qiyun Zhu (qzhu@jcvi.org)
Affiliation: J. Craig Venter Institute
Last updated: Jan 28, 2017

Usage:
  python scripts/databaser.py [-format=none|blast] [-range=all|microbe|etc] [-represent=no|auto|filename] [-subsample=0|1|2...] [-out=dbname]
Note:
  By default, the script will download NCBI RefSeq proteomic data of microbial organisms, keep one proteome per species, plus all NCBI-defined representative proteomes.
  Use "-help" to display details of program parameters.
Press Enter to continue or Ctrl+C to exit...
Connecting to the NCBI FTP server... done.
Downloading the NCBI taxonomy database... done.
Reading the NCBI taxonomy database... done.
  1591763 taxa read.
Downloading the NCBI representative genome list... done.
Reading the NCBI representative genome list... done.
  4380  genomes read.
Reading RefSeq genome list... done.
  85021 genomes read.
Subsampling genomes... done.
  13379 genomes retained.
Downloading non-redundant proteomic data from NCBI RefSeq...
  25 Traceback (most recent call last):
  File "databaser.py", line 292, in <module>
    files = ftp.nlst()
  File "/usr/lib/python2.7/ftplib.py", line 506, in nlst
    self.retrlines(cmd, files.append)
  File "/usr/lib/python2.7/ftplib.py", line 429, in retrlines
    conn = self.transfercmd(cmd)
  File "/usr/lib/python2.7/ftplib.py", line 368, in transfercmd
    return self.ntransfercmd(cmd, rest)[0]
  File "/usr/lib/python2.7/ftplib.py", line 327, in ntransfercmd
    conn = socket.create_connection((host, port), self.timeout)
  File "/usr/lib/python2.7/socket.py", line 571, in create_connection
    raise err
socket.error: [Errno 110] Connection timed out

I don't know how to resolve the problem, could you help me? please!
thank you!!

**Comment by qiyunzhu** [comment]
Hi @germant13 Thanks for letting me know! It is likely a network setting problem. Can you try another computer / network service? For example, run it from you laptop thru your home wifi. Hope that will solve the problem!

**Comment by germant13** [comment]
Thank you for the response
I have a new problem. I have tried twice time and obtained the same error. I have used two computer (laptop and PC computer) and , also, I am using to LAN connecttion in my home.
It show the error like that:
germanmtraglia@ubuntu:/media/germanmtraglia/6E5CA6935CA6561D/seq/new_seq_robert/HGTector-master/scripts$ python databaser.py 

Databaser: generate taxonomically balanced, non-redundant proteome database.
Author: Qiyun Zhu (qzhu@jcvi.org)
Affiliation: J. Craig Venter Institute
Last updated: Jan 28, 2017

Usage:
  python scripts/databaser.py [-format=none|blast] [-range=all|microbe|etc] [-represent=no|auto|filename] [-subsample=0|1|2...] [-out=dbname]
Note:
  By default, the script will download NCBI RefSeq proteomic data of microbial organisms, keep one proteome per species, plus all NCBI-defined representative proteomes.
  Use "-help" to display details of program parameters.
Press Enter to continue or Ctrl+C to exit...
Connecting to the NCBI FTP server... done.
Downloading the NCBI taxonomy database... done.
Reading the NCBI taxonomy database... done.
  1593970 taxa read.
Downloading the NCBI representative genome list... done.
Reading the NCBI representative genome list... done.
  4380  genomes read.
Reading RefSeq genome list... done.
  85728 genomes read.
Subsampling genomes... done.
  13465 genomes retained.
Downloading non-redundant proteomic data from NCBI RefSeq...
  25 50 75 100 125 150 175 200 225 250 275 300 325 350 375 400 425 450 475 500 525 550 575 600 625 650 675 700 725 750 775 800 825 850 875 900 925 950 975 1000 1025 1050 1075 1100 1125 1150 1175 1200 1225 1250 1275 1300 1325 1350 1375 1400 1425 1450 1475 1500 1525 1550 1575 1600 1625 1650 1675 1700 1725 1750 1775 1800 1825 1850 1875 1900 1925 1950 1975 2000 2025 2050 2075 2100 2125 2150 2175 2200 2225 2250 2275 2300 2325 2350 2375 2400 2425 2450 2475 2500 2525 2550 2575 2600 2625 2650 2675 2700 2725 2750 2775 2800 2825 2850 2875 2900 2925 2950 2975 3000 3025 3050 3075 3100 3125 3150 3175 3200 3225 3250 3275 3300 3325 3350 3375 3400 3425 3450 3475 3500 3525 3550 3575 3600 3625 3650 3675 3700 3725 3750 3775 3800 3825 3850 3875 3900 3925 3950 3975 4000 4025 4050 4075 4100 4125 4150 4175 4200 4225 4250 4275 4300 4325 4350 4375 4400 4425 4450 4475 4500 4525 4550 4575 4600 4625 4650 4675 4700 4725 4750 4775 4800 4825 4850 4875 4900 4925 4950 4975 5000 5025 5050 5075 5100 5125 5150 5175 5200 5225 5250 5275 5300 5325 5350 5375 5400 5425 5450 5475 5500 5525 5550 5575 5600 5625 5650 5675 5700 5725 5750 5775 5800 5825 5850 5875 5900 5925 5950 5975 6000 6025 6050 6075 6100 6125 6150 6175 6200 6225 6250 6275 6300 6325 Traceback (most recent call last):
  File "databaser.py", line 380, in <module>
    gi2taxids[gi] = taxid
MemoryError

Regards

German
